robocopkaka/event-manager-rails
# #165310625 Add methods to fetch events

#### What does this PR do?
Allows users to retrieve events
#### Description of Task to be completed?
* Allow users fetch all events
* Allow users fetch upcoming events
* Allow users fetch a single event
#### How should this be manually tested?
To fetch all events
```
GET http://base_url/api/v1/centers/:id/events
```
Should return
```.json
{
    "message": "Successfully retrieved events",
    "data": {
        "events": [
            {
                "id": 1,
                "name": "Drag race 1",
                "description": "Drag races and shit",
                "guests": 100,
                "start_time": "2019-02-02T13:00:00.000Z",
                "end_time": "2019-02-02T20:24:18.544Z"
            },
            {
                "id": 2,
                "name": "Drag race 2",
                "description": "Drag races and shit",
                "guests": 100,
                "start_time": "2019-02-02T13:00:00.000Z",
                "end_time": "2019-02-02T20:24:18.544Z"
            }
        ]
    }
}
```

To fetch upcoming events
```
GET http://base_url/api/v1/centers/:id/events?upcoming=true
```
Should return
```.json
{
    "message": "Successfully retrieved events",
    "data": {
        "events": [
            {
                "id": 1,
                "name": "Drag race 1",
                "description": "Drag races and shit",
                "guests": 100,
                "start_time": "2019-02-02T13:00:00.000Z",
                "end_time": "2019-02-02T20:24:18.544Z"
            },
            {
                "id": 2,
                "name": "Drag race 2",
                "description": "Drag races and shit",
                "guests": 100,
                "start_time": "2019-02-02T13:00:00.000Z",
                "end_time": "2019-02-02T20:24:18.544Z"
            }
        ]
    }
}
```

To fetch a single event
```
GET http://localhost:3001/api/v1/centers/:center_id/events/:event_id
```

Should return:
```.json
{
    "message": "Event retrieved successfully",
    "data": {
        "event": {
            "id": 1,
            "name": "Drag race 1",
            "description": "Drag races and shit",
            "guests": 100,
            "start_time": "2019-02-02T13:00:00.000Z",
            "end_time": "2019-02-02T20:24:18.544Z"
        }
    }
}
```

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories? (if applicable)
[#165310625](https://www.pivotaltracker.com/story/show/165310625)
